app/policies/application_policy.rb

```diff
@@ -0,0 +1,73 @@
+class ApplicationPolicy
+  attr_reader :user, :context, :params
+
+  def initialize(context, record)
+    @context = context
+    @record = record
+  end
+
+  delegate :user, to: :context
+  delegate :expert, to: :context
+
+  def index?
+    true
+  end
+
+  def show?
+    false
+  end
+
+  def create?
+    true
+  end
+
+  def new?
+    create?
+  end
+
+  def update?
+    false
+  end
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    false
+  end
+
+  def support?(user, diagnosis)
+    if user.is_a?(User)
+      user.experts.each do |expert|
+        return true if expert.experts_subjects.support_for(diagnosis).present?
+      end
+    end
+    false
```

Comment:
En principe, `user` est soit `nil`, soit effectivement un `User`, donc on pourrait simplement écrire `user.present?`.
En fait, la méthode peut se réécrire, avec le _safe navigation operator_ `&.` et `Enumerable#any?`
```suggestion
    user&.experts.any? do |expert|
      expert.experts_subjects.support_for(diagnosis).present?
    end
```